projects/plugins/jetpack/tests/php/modules/csstidy/test-class.jetpack-csstidy.php

```diff
@@ -0,0 +1,63 @@
+<?php
+/**
+ * Class WP_Test_Jetpack_CSSTidy.
+ *
+ * @package automattic/jetpack
+ */
+
+require_jetpack_file( 'modules/custom-css/csstidy/class.csstidy.php' );
+
+/**
+ * Class WP_Test_Jetpack_CSSTidy
+ */
+class WP_Test_Jetpack_CSSTidy extends WP_UnitTestCase {
+
+	/**
+	 * The tested instance.
+	 *
+	 * @var csstidy
+	 */
+	public $instance;
+
+	/**
+	 * Sets up each test.
+	 *
+	 * @inheritDoc
+	 */
+	public function setUp() {
+		parent::setUp();
+		$this->instance = new csstidy();
+		$this->instance->set_cfg( 'optimise_shorthands', 0 );
+	}
+
+	/** Provides values for CSS preserve leading zero patterns */
+	public function custom_preserve_leading_zeros_provider() {
+		// phpcs:ignore Squiz.PHP.CommentedOutCode.Found -- false positive
+		// 'test case description' => [ 'input', 'expected output', 'preserve_leading_zero' ].
+		return array(
+			'test_removes_leading_zeros_by_default_single_value' => array( 'marquee {line-height:0.7}', "marquee {\nline-height:.7\n}", false ),
+			'test_decimals_greater_than_zero_unchanged'    => array( 'blink {line-height:1.7}', "blink {\nline-height:1.7\n}", false ),
+			'test_removes_leading_zeros_by_default_multiple_values' => array( 'dfn {margin-left:-0.7px;top:0.55rem;line-height:0.3333;}', "dfn {\nmargin-left:-.7px;\ntop:.55rem;\nline-height:.3333\n}", false ),
+			'test_preserves_leading_zeros_single_value'    => array( 'aside {line-height:0.7}', "aside {\nline-height:0.7\n}", true ),
+			'test_preserves_leading_zeros_multiple_values' => array( 'code {margin-left:-0.7px;top:0.55rem;line-height:0.3333;}', "code {\nmargin-left:-0.7px;\ntop:0.55rem;\nline-height:0.3333\n}", true ),
```

Comment:
This test is failing. Not sure if it's the units or the length of the string. Have to check.

```bash
--- Expected
+++ Actual
@@ @@
 'code {\n
-margin-left:-0.7px;\n
+margin-left:-.7px;\n
 top:0.55rem;\n
 line-height:0.3333\n
 }'
```

I think it's happening somewhere after we add the settings check in https://github.com/Automattic/jetpack/pull/19993/files#diff-98d7dd2b1fbcfdea33583cead4221376c939b2180e1dedaf6f20d5eeb55feb9eR407